aea/registries/base.py

```diff
@@ -28,12 +28,13 @@
 from abc import ABC, abstractmethod
```

Comment:
this is another big change of the PR. The `Registry` class and its subclasses have been refactored, by improving modularity and code reuse.